app/oauth.go

```diff
@@ -477,7 +477,12 @@ func (a *App) RegenerateOAuthAppSecret(app *model.OAuthApp) (*model.OAuthApp, *m
 
 func (a *App) RevokeAccessToken(token string) *model.AppError {
 	session, _ := a.GetSession(token)
-	schan := a.Srv.Store.Session().Remove(token)
+	schan := make(chan *model.AppError, 1)
+	go func() {
+		removeErr := a.Srv.Store.Session().Remove(token)
+		schan <- removeErr
```

Comment:
Nit: You don't need the extra variable:
```suggestion
		schan <-  a.Srv.Store.Session().Remove(token)
```